system-test/pubsub.ts

```diff
@@ -1023,6 +1031,153 @@ describe('pubsub', () => {
+      ).toString();
+
+      return schemaDef;
+    };
+
+    const setupTestSchema = async () => {
+      const schemaDef = await getSchemaDef();
+      const schemaId = generateSchemaName();
+      await pubsub.createSchema(schemaId, SchemaTypes.Avro, schemaDef);
+      return schemaId;
+    };
+
+    it('should create a schema', async () => {
+      const schemaId = await setupTestSchema();
+      const schemaList = await aiToArray(pubsub.listSchemas(), schemaId);
+      assert.strictEqual(schemaList.length, 1);
+    });
+
+    it('should delete a schema', async () => {
+      const schemaId = await setupTestSchema();
+
+      // Validate that we created one, because delete() doesn't throw, and we
+      // might end up causing a false negative.
+      const preSchemaList = await aiToArray(pubsub.listSchemas(), schemaId);
+      assert.strictEqual(preSchemaList.length, 1);
+
+      await pubsub.schema(schemaId).delete();
+
+      const postSchemaList = await aiToArray(pubsub.listSchemas(), schemaId);
+      assert.strictEqual(postSchemaList.length, 0);
+    });
+
+    it('should list schemas', async () => {
+      const schemaId = await setupTestSchema();
+
+      const basicList = await aiToArray(
+        pubsub.listSchemas(SchemaViews.Basic),
+        schemaId
+      );
+      assert.strictEqual(basicList.length, 1);
+      assert.strictEqual(basicList[0].definition, '');
+
+      const fullList = await aiToArray(
+        pubsub.listSchemas(SchemaViews.Full),
+        schemaId
+      );
+      assert.strictEqual(fullList.length, 1);
+      assert.ok(fullList[0].definition);
+    });
+
+    it('should get a schema', async () => {
+      const schemaId = await setupTestSchema();
+      const schema = pubsub.schema(schemaId);
+      const info = await schema.get(SchemaViews.Basic);
+      assert.strictEqual(info.definition, '');
+
+      const fullInfo = await schema.get(SchemaViews.Full);
+      assert.ok(fullInfo.definition);
+    });
+
+    it('should validate a schema', async () => {
+      const schemaDef = await getSchemaDef();
+
+      try {
+        await pubsub.validateSchema({
+          type: SchemaTypes.Avro,
+          definition: schemaDef,
+        });
+      } catch (e) {
+        assert.strictEqual(e, undefined, 'Error thrown by validateSchema');
+      }
+
+      const badSchemaDef = '{"not_actually":"avro"}';
+      try {
+        await pubsub.validateSchema({
+          type: SchemaTypes.Avro,
+          definition: badSchemaDef,
+        });
+      } catch (e) {
+        assert.ok(e);
+      }
+
+      const fakeSchemaDef = 'woohoo i am a schema, no really';
+
+      try {
+        await pubsub.validateSchema({
+          type: SchemaTypes.Avro,
+          definition: fakeSchemaDef,
+        });
+      } catch (e) {
+        assert.ok(e);
+      }
+    });
+
+    // The server doesn't seem to be returning proper responses for this.
+    // Commenting out for now, until it can be discussed.
```

Comment:
by convention, I would put this as a:

```
//  TODO(feywind): uncomment this test
```

In the future, I think we might even automatically open issues when folks do this.